src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryCachingConfiguration.java

```diff
@@ -53,7 +53,12 @@ public String getExcludedVersionsStr() {
     }
 
     public Boolean isExcluded(String version) {
-        return getExcludedVersions().contains(version);
+        for (String it : getExcludedVersions()){
+            if (version.contains(it)){
```

Comment:
This line is why `LibraryCachingConfigurationTest.isExcluded` is failing with an NPE: `version` may be null.

It's also why `ResourceStepTest.caching` is failing: when `excludedVersionsStr` is `""`, `getExcludedVersions()` returns a set with a single `""`, and `version.contains("")` is true for every non-null version.

Reply:
@dwnusbaum dwnusbaum, PR should be ready for review